src/js/productDetails.mjs

```diff
@@ -11,8 +11,11 @@ export default async function productDetails(productId) {
   // once the HTML is rendered we can add a listener to Add to Cart button
   document.getElementById("addToCart").addEventListener("click", () => addToCart(product));
 }
+
 function addToCart(product) {
-  setLocalStorage("so-cart", product);
+  const currentCart = getLocalStorage("so-cart") || [];
+  currentCart.push(product);
+  setLocalStorage("so-cart", currentCart);
```

Comment:
@nicholas-28 @Alitares27 @WilsonGbedemah You see that previously we were just setting the localStorage equal to the one product that was added to the cart. However, throughout the codebase it wants the cart to be an array.

So the fix here is to get the current cart with `getLocalStorage("so-cart")` OR if nothing is set there we default to an empty array `[]`.

Once we have the current cart, we `push` the new product we want to add to the cart. So now we have an array of products--ones that existed previously (if any) AND the new one.

After we have added it to the array locally in this function we then want to save the results to localStorage so they persist and we do that.